sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/stringExpressions.scala

```diff
@@ -204,6 +204,29 @@ case class Elt(children: Seq[Expression])
       }
     }
   }
+
+  override protected def doGenCode(ctx: CodegenContext, ev: ExprCode): ExprCode = {
+    val index = children.head.map(_.genCode(ctx))(0)
+    val strings = children.tail.map(_.genCode(ctx))
+    val stringValues = strings.map { eval =>
+      s"${eval.isNull} ? null : ${eval.value}"
+    }.mkString(", ")
+    val indexVal = ctx.freshName("index")
+    val stringArray = ctx.freshName("strings");
+
+    ev.copy(index.code + "\n" + strings.map(_.code).mkString("\n") + s"""
+      int $indexVal = ${index.value} - 1;
+      UTF8String[] $stringArray = {$stringValues};
```

Comment:
Let me do a simple benchmark for this. Actually I think it is no harm to have codegen for this.